src/frontend/src/common/components/library/data_subview/components/EditSamplesConfirmationModal/components/ImportFile/index.tsx

```diff
@@ -167,13 +172,34 @@ export default function ImportFile({
             | NamedGisaidLocation
             | undefined) = passOrDeleteEntry(value);
         }
+        const filledInUploadedMetadata = {
+          ...pick(uploadedMetadataEntry, uploadedFieldsWithData),
+        };
+
+        if (!isEmpty(uploadedMetadataEntry)) {
```

Comment:
This comment isn't specific to this line, but the size of this overall for-loop feels like it's starting to get out of hand (line 142, `for (const sampleId of Object.keys(metadata))`). I'm just generally finding it kind of hard to reason abut what's going on with this much happening in one place. I know you've been working on the Edit Metadata for awhile, so I'm not pushing for it to happen here and now, but generally I'd like to chop things up into smaller functional pieces so it's easier to reason about what's going on.

Reply:
i'll see what i can do